src/hypothesis/strategies.py

```diff
@@ -982,32 +987,61 @@ def fractions(min_value=None, max_value=None, max_denominator=None):
 
     If max_denominator is not None then the absolute value of the denominator
     of any generated values is no greater than max_denominator. Note that
-    max_denominator must be at least 1.
+    max_denominator must be None or a positive integer, and the denominator
```

Comment:
> Having fixed that, I actually can't determine in the general case whether there are any legal values when the value bounds are at higher precision than allowed for examples. I'd be very happy to be proved wrong here!

Maybe I don't understand the question, but I think this is easy isn't it? e.g. `fractions(min_value=Fraction(1, 3), max_value=Fraction(2, 3), max_denominator=2)` has the valid value `Fraction(1, 2)`.